bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ClientConfiguration.java

```diff
@@ -145,14 +145,6 @@
     // Client auth provider factory class name. It must be configured on Bookies to for the Auditor
     protected static final String CLIENT_AUTH_PROVIDER_FACTORY_CLASS = "clientAuthProviderFactoryClass";
 
-    // Client TLS
-    protected static final String TLS_KEYSTORE_TYPE = "clientKeyStoreType";
```

Comment:
this breaks BC. 

for each field you are change here, I think you have to do:

- in this file: change "TLS_XYZ" to "CLIENT_XYZ", but not change the value.
- change getTLSxyz() to : getProperty(TLS_XYZ, getProperty(CLIENT_XYZ, ...))

I am not what AutoRecovery is using, is it using client settings or server settings? if it is using client settings, you need also to update the configuration settings in the bookie conf.

Reply:
I am moving TLS_XYZ from ServerConfiguration and ClientConfiguration to the common class AbstractConfiguration and just use one set of config params (no clientXYZ params anymore). So, effectively this is a dead code. What I am doing here is to keep just one set of config params so both bookie and client config params names are same.

AutoRecovery can still use the same set of params as bookies to configure the bookkeeper client.

Only config param names are changing, but the variable name still remains the same as TLS_XYZ. Do you still think its a BC issue?

Reply:
yeah param name changing is considered as breaking BC. because if a client upgrades from 4.6.0 to 4.7.0, without changing the configuration, it won't work.